### ISSUE TITLE
Nomenclature

### ISSUE THREAD

**Issue by aimalz** [LSSTDESC/CLMM#2]
We need more descriptive yet still short names for the structures we're thinking about.  Please use this space to discuss.

**Comment by aimalz** [comment]
While we're working on #19, we might want to settle on some solution to #2 before we build up too much that a linter can't fix it.

**Comment by m-aguena** [comment]
Is this regarding the general format of names we are going to use or do you (@aimalz) have some specific examples in mind?

**Comment by m-aguena** [comment]
I thought we decided on classes being written with words beginning with caps and no spaces like `ThisExampleClassName` and functions would be written in all lower cases with words separated by `_` like `this_example_function`.

**Comment by matthewkirby** [comment]
I think the discussion was about what the particular names were rather than
how they should be formatted.

**Comment by m-aguena** [comment]
Ok then, lets get specific. What are the bad names we have implemented so far?

**Comment by aimalz** [comment]
I think at the meeting we'd sorted out that there were issues with `model`, `profile`, `parameter`, `data`, and `catalog`.  I'm struggling to figure out which of these have been addressed and in which branches. . .

**Comment by matthewkirby** [comment]
I think then it might be best to just sort it out here, and then when we are reviewing PRs we make sure they adhere to that. 

The current status, as far as I know

- `model` is unchanged everywhere
- `profile` we don't have anything called this. I believe it was called `Profile1D` in the Issue/16 branch. I am seeing that the NFWProfile class is inheriting Profile1D but the class that is written is called RadialProfile1D.
- `parameter` this is just a dictionary for the moment and nothing more is written.
-`data` unsure
- `catalog` unsure

**Comment by aimalz** [comment]
I think right now everything that could be construed as data falls under the generic `GCData` object, with keys that have to match between the data and whatever wants to interact with that, but maybe another name would be more appropriate.  I think `Parameter` objects might end up being similar to `GCData` objects, either by both becoming more like the [lmfit](https://lmfit.github.io/lmfit-py/parameters.html#lmfit.parameter.Parameter) version (or something we cook up ourselves, whatever makes it easier to include (multivariate) priors).  Are we satisfied with the `GCData` and `Parameter` names?  I think `Parameter` is suboptimal due to overlap with our documentation style, so we should consider an alternative.  And improvements to `GCData` wouldn't be unwelcome either.

I think right now everything that isn't data or parameters is a model, so we've got `models.RadialProfile`/`models.Profile1D` and, presumably, something for mass-concentration relations, etc.  I don't really know how these things are used in inference and thus what will make sense to a user, but now would be a good time to sort out if `Model` subclasses are distinguished by the physical quantities involved or just the math -- I think the former might make more sense, because `clmm` doesn't need to be the most general package ever, just something comprehensive for population-level galaxy cluster studies.  Do we want a `Model` subclass that only includes radial dark matter density models like NFW or are there other quantities that we'd want as radial functions?  When we add a mass-concentration model, do we want it to know that it has a physical meaning or just be a function of two variables?  This sort of ties into the weak lensing utilities file -- what kinds of things are intended to end up there, and how will they be used?  I want to make sure our objects/functions have descriptive names, i.e. not `Kappa` or `Gamma` but perhaps `LensingKernel` or `ShearMap`.

Are we okay with `Manager` sticking around?  (Does that conflict with any really basic internal Python stuff?)  And are we okay with `Ensemble` being the object that contains multiple `GalaxyCluster` objects and facilitates the `Manager` interacting with them?  (I think I mentioned `Catalog` mostly to prevent myself from incorporating it into `GalaxyCluster` or `Ensemble` objects.)
